NitroxServer/GameLogic/Entities/EntityData.cs

```diff
@@ -67,7 +67,10 @@ public void AddEntity(Entity entity)
             {
                 lock (globalRootEntitiesByGuid)
                 {
-                    globalRootEntitiesByGuid.Add(entity.Guid, entity);
+                    if (!globalRootEntitiesByGuid.ContainsKey(entity.Guid))
```

Comment:
Although this is technically correct (and nice to have safety) I think it is masking a problem.  We shouldn't be trying to add an item to the global root if it is already there.  This probably points to another problem (maybe we aren't correctly removing the item from this dict when we pick up items).

Reply:
Ok no worries , I'll take a quick look at how its being handled and see if i can change it to be removed on pickup.